src/Features/Core/Portable/ExtractClass/AbstractExtractClassRefactoringProvider.cs

```diff
@@ -55,28 +58,38 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
 
         private async Task<ExtractClassWithDialogCodeAction?> TryGetMemberActionAsync(CodeRefactoringContext context, IExtractClassOptionsService optionsService)
         {
-            var selectedMemberNode = await GetSelectedNodeAsync(context).ConfigureAwait(false);
-            if (selectedMemberNode is null)
+            var selectedMemberNodes = await GetSelectedNodesAsync(context).ConfigureAwait(false);
+            if (selectedMemberNodes.IsEmpty)
             {
                 return null;
             }
 
             var (document, span, cancellationToken) = context;
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var selectedMember = semanticModel.GetDeclaredSymbol(selectedMemberNode, cancellationToken);
-            if (selectedMember is null || selectedMember.ContainingType is null)
+            var memberNodeSymbolPairs = selectedMemberNodes
+                .Select(m => (node: m, symbol: semanticModel.GetDeclaredSymbol(m, cancellationToken)))
+                // Use same logic as pull members up for determining if a selected member
+                // is valid to be moved into a base
+                .Where(pair => pair.symbol != null && MemberAndDestinationValidator.IsMemberValid(pair.symbol))
+                .AsImmutable();
+
+            var selectedMembers = memberNodeSymbolPairs.SelectAsArray(pair => pair.symbol!);
+
+            if (selectedMembers.IsEmpty)
             {
                 return null;
             }
 
-            // Use same logic as pull members up for determining if a selected member
-            // is valid to be moved into a base
-            if (!MemberAndDestinationValidator.IsMemberValid(selectedMember))
+            var containingType = selectedMembers.First().ContainingType;
+            if (containingType == null || selectedMembers.Any(m => m.ContainingType != containingType))
```

Comment:
```suggestion
            var containingType = selectedMembers.First().ContainingType;
            Contract.ThrowIfNull(containingType);
            if (selectedMembers.Any(m => !containingType.Equals(m.ContainingType)))
```

Symbols cannot be tested for equality using reference-equality.  the compiler is free to create other instances that are .Equals but are not `==`.